### ISSUE TITLE
Issue when trying to install cider 20140912.503

### ISSUE THREAD

**Issue by bretthoerner** [melpa/melpa#2005]
I have 20140905.2016 installed, for what it's worth. Trying to upgrade (for the last few days)...

```
Debugger entered--Lisp error: (error "Package `queue-0.1.1' is unavailable")
  signal(error ("Package `queue-0.1.1' is unavailable"))
  error("Package `%s-%s' is unavailable" "queue" "0.1.1")
  package-compute-transaction((cider) ((clojure-mode (2 0 0)) (cl-lib (0 3)) (dash (2 4 1)) (pkg-info (0 4)) (emacs (24)) (queue (0 1 1))))
  package-install(cider)
  mapc(package-install (cider))
  package-menu-execute()
  call-interactively(package-menu-execute nil nil)
```

**Comment by samaaron** [comment]
I just bumped into this issue too. I'm using Emacs 26.2 and my `init.el` looks as follows:

```
(require 'package)

(let* ((no-ssl (and (memq system-type '(windows-nt ms-dos))
                    (not (gnutls-available-p))))
       (proto (if no-ssl "http" "https")))
  (when no-ssl (warn "\
Your version of Emacs does not support SSL connections,
which is unsafe because it allows man-in-the-middle attacks.
There are two things you can do about this warning:
1. Install an Emacs version that does support SSL and be safe.
2. Remove this warning from your init file so you won't see it again."))
  ;; Comment/uncomment these two lines to enable/disable MELPA and MELPA Stable as desired
  ;;(add-to-list 'package-archives (cons "melpa" (concat proto "://melpa.org/packages/")) t)
  (add-to-list 'package-archives (cons "melpa-stable" (concat proto "://stable.melpa.org/packages/")) t)
  (add-to-list 'package-archives (cons "gnu" (concat proto "://elpa.gnu.org/packages/")))
  (when (< emacs-major-version 24)
    ;; For important compatibility libraries like cl-lib
    (add-to-list 'package-archives (cons "gnu" (concat proto "://elpa.gnu.org/packages/")))))

(package-initialize)

(unless package-archive-contents
  (package-refresh-contents))

(setq user-full-name "Sam Aaron"
      user-mail-address "samaaron@gmail.com")

(unless (package-installed-p 'use-package)
  (package-install 'use-package))

(unless (package-installed-p 'cider)
  (package-install 'cider))
```

I'm seeing: `error: Package ‘queue-0.2’ is unavailable`

**Comment by samaaron** [comment]
Seems like my issue was some SSL conflict - perhaps with a locally installed OpenSSL.

Reading this gave a suggestion of disabling TLS1.3 support
https://debbugs.gnu.org/cgi/bugreport.cgi?bug=34341

Which can be achieved with:

```
(setq gnutls-algorithm-priority "NORMAL:-VERS-TLS1.3")
```

I'm not fully aware of the security implications of this, so can't recommend anyone else do it, but it al least got me over my hurdle.

**Comment by pwalsh** [comment]
I'm seeing this too.

**Comment by pwalsh** [comment]
After setting `(setq gnutls-algorithm-priority "NORMAL:-VERS-TLS1.3")` I still see `Error (use-package): Failed to install cider: Package ‘queue-0.2’ is unavailable`

**Comment by milkypostman** [comment]
Seems like maybe the queue dependency is not available? Is this melpa
stable?

**Comment by ackerleytng** [comment]
I have both melpa stable and melpa and I'm seeing this too.

**Comment by Macroz** [comment]
Got into this issue in a fresh installation of Xubuntu 19.04 and the TLS trick did it for me. Apparently there is a fix already see https://www.reddit.com/r/emacs/comments/cdei4p/failed_to_download_gnu_archive_bad_request/ and https://debbugs.gnu.org/cgi/bugreport.cgi?bug=34341.

**Comment by stevematdavies** [comment]
Still having this exact issue Emacs on Mac 
26.2

**Comment by peterhhchan** [comment]
**tldr;**
I had to downgrade my TLS  version to 1.2.
`(setq gnutls-algorithm-priority "NORMAL:-VERS-TLS1.2")`

**longer version**
I encountered this issue when setting up emacs 26.3 on windows for the first time.  FWIW, my issue wasn't related to cider or melpa, though I am using both.  Rather, it was an issue with the gnutls libraries that came packaged with my emacs binaries.

To check if its a TLS issue, simply change `"https://melpa.org/packages/"` to `"http://melpa.org/packages/"`.  If everything starts working, it probably means you have a problem with your protocol libraries.  Obviously, its tempting to stop here, but using an older version of TLS is preferable to no TLS at all.  So:

`(setq gnutls-algorithm-priority "NORMAL:-VERS-TLS1.2")`

**Comment by purcell** [comment]
Yes, I include the following workaround for broken TLS support in my personal config, and others might find this snippet helpful too: https://github.com/purcell/emacs.d/blob/71db52615191b88b997a795119ae9a3318a4f243/lisp/init-elpa.el#L27-L29
